bbn-test-5/finality-provider/upgrades/v2rc0/README.md

```diff
@@ -0,0 +1,190 @@
+    signatures not utilizing context signing will be rejected, and **prevent
+    the Babylon Genesis Finality Provider from submitting a valid finality
+    signature for that height**. The reason is that each signature utilizes
+    a public randomness commitment of a specific height, which can't be re-used.
+  - Thus, it's critical that the Babylon Genesis Finality Providers upgrade
+    their software and properly configure the context signing height no later
+    than the Babylon Genesis Testnet upgrade height `1692200`.
+    **For safety, Finality Provider operators must upgrade
+    their software earlier than the Babylon Genesis Testnet network upgrade.**
+- Major refactoring of the codebase into a reusable SDK. This aims to enable
+  the development of Finality Provider software for Bitcoin Supercharged
+  Networks (BSNs). The initial BSN stacks will include Ethereum L2 Rollups and
+  Cosmos SDK chains.
+  - It's important to emphasize that the BSN Finality Provider software will be
+    distinct from the Babylon Finality Provider software, coming with dedicated
+    binaries and operational guides. Babylon Finality Providers will continue
+    operating the original software.
+
+## 3. Applying the upgrade
+
+Applying the upgrade constitutes of 2 steps:
+- Upgrade your Finality Provider and EOTS daemons **before** the Babylon Genesis
+  Testnet upgrade height `1692200` is reached.
+- Upgrade your Babylon node once the network halts.
+
+### 3.1. Upgrade the Finality Provider
+
+**NOTE: THIS MUST HAPPEN BEFORE THE BABYLON GENESIS TESTNET UPGRADE HEIGHT
+`1692200` IS REACHED!**
+
+#### 3.1.1. Preparation
+
+Follow the steps below:
+- Ensure that your Babylon Finality Provider and EOTS daemons are currently
+  operating version `v1.0.0`, `v1.1.0-rc.0` or `v1.1.0-rc.1`.
+- Obtain the `v2.0.0-rc.0` binary. You can achieve this in multiple ways:
+  - Download the binary from the [releases
+    page](https://github.com/babylonlabs-io/finality-provider/releases/tag/v2.0.0-rc.0).
+  - Build the binary on your machine:
+    ```shell
+    git checkout v2.0.0-rc.0
+    make install
+    ```
+  - Pull the pre-built Docker image:
+    ```shell
+    docker pull babylonlabs/finality-provider:v2.0.0-rc.0
+    ```
+
+#### 3.1.2. Execution
+
+The upgraded Babylon Genesis Finality Provider and EOTS daemon binaries are
+tightly coupled with the Babylon Genesis Testnet software upgrade.
+
+The following steps must be executed **before the Babylon Genesis Testnet
+upgrade height is reached**, to ensure that your finality provider does not
+experience downtime or submit invalid signatures:
+1. **Stop the Finality Provider and EOTS daemons**.
+2. **Swap the Babylon Genesis finality provider and EOTS daemon binaries with
+   the new ones**.
+3. In your Babylon Genesis Finality Provider config, append the following
+   configuration to the `[Application Options]` section:
+   ```shell
+   [Application Options]
+
```

Comment:
Found some other updates in the config as well
```
; The upper bound of the number of Schnorr public randomness for each commitment
NumPubRandMax = 500000

; The interval between each attempt to commit public randomness
RandomnessCommitInterval = 30s
```

Also this config is removed in v2
```
; the type of the consumer chain
ChainType = babylon
```